pkg/ebpf/c/common/context.h

```diff
@@ -22,49 +23,61 @@ init_context(void *ctx, event_context_t *context, struct task_struct *task, u32
 {
     long ret = 0;
     u64 id = bpf_get_current_pid_tgid();
-    context->task.start_time = get_task_start_time(task);
+
+    // NOTE: parent is always a real process, not a potential thread group leader
+    struct task_struct *leader = get_leader_task(task);
+    struct task_struct *up_parent = get_leader_task(get_parent_task(leader));
+
+    // Task Info on Host
     context->task.host_tid = id;
     context->task.host_pid = id >> 32;
-    context->task.host_ppid = get_task_ppid(task);
+    context->task.host_ppid = get_task_pid(up_parent); // always a real process (not a lwp)
+    // Namespaces Info
     context->task.tid = get_task_ns_pid(task);
     context->task.pid = get_task_ns_tgid(task);
-    context->task.ppid = get_task_ns_ppid(task);
+    context->task.ppid = get_task_ns_pid(up_parent); // ns pid for up_parent
```

Comment:
This is not necessarily the true value here.
PID 1 of a PID namespace have no parent in its namespace.
This results that we get the PID of the parent under its own namespace.
Unless we want this behavior, I would have expected to get PPID=0 in this case.
WDYT?